tbx/static_src/javascript/components/youtube-embed.js

```diff
@@ -0,0 +1,79 @@
+import Cookies from 'js-cookie';
+
+/**
+ * YouTubeConsentManager handles the consent for loading YouTube videos on a webpage.
+ */
+class YouTubeConsentManager {
+    static selector() {
+        return '.grid__embed.streamfield__embed.youtube-video-container';
+    }
+
+    /**
+     * Create a new YouTubeConsentManager.
+     */
+    constructor(node) {
+        this.consentButtonClass = 'consent-button';
+        this.dontAskAgainButtonClass = 'dont-ask-again-button';
+        this.videoPlaceholderClass = 'youtube-video-placeholder';
+        this.youtubeEmbedClass = 'youtube-embed';
+
+        this.youtubeEmbedContainer = node;
+        this.consentButton = this.youtubeEmbedContainer.querySelector(
+            `.${this.consentButtonClass}`,
```

Comment:
Again we would ideally keep the classes for styling, and use data attributes for selection. I'm happy to figure this out as part of the front-end work so I'm not too worried if this is merged as is for now.